google-cloud-storage/src/main/java/com/google/cloud/storage/UnifiedOpts.java

```diff
@@ -1070,6 +1075,19 @@ public Mapper<RewriteObjectRequest.Builder> rewriteObject() {
     }
   }
 
+  static final class MatchGlob extends RpcOptVal<String> implements ObjectListOpt {
+    private static final long serialVersionUID = 8819855597395473178L;
+
+    private MatchGlob(String val) {
+      super(StorageRpc.Option.MATCH_GLOB, val);
+    }
+
+    @Override
+    public Mapper<ListObjectsRequest.Builder> listObjects() {
+      throw new UnsupportedOperationException("Option matchGlob is not supported for gRPC yet.");
```

Comment:
We have a helper method to construct an appropriate exception with a pointer on how they can use the method.

```suggestion
      return GrpcStorageImpl.throwHttpJsonOnly(
          com.google.cloud.storage.Storage.BlobListOption.class, "matchGlob(String)");
```

Using this, the exception will have a message of
```
java.lang.UnsupportedOperationException: com.google.cloud.storage.Storage$BlobListOption#matchGlob(String) is only supported for HTTP_JSON transport. Please use StorageOptions.http() to construct a compatible instance.
```